prowler

```diff
@@ -441,10 +445,13 @@ if [[ $CHECK_ID ]];then
 fi
 
 if [[ $ACCOUNT_TO_ASSUME ]]; then
-  # unset env variables with assumed role credentials
-  unset AWS_ACCESS_KEY_ID
-  unset AWS_SECRET_ACCESS_KEY
-  unset AWS_SESSION_TOKEN
+      #This would remove permissions if all checks were run using external ID
```

Comment:
This if `if [[ -z $ROLE_EXTERNAL_ID ]]` would make to not to clean up those variables unless external ID is created so it is not needed.

Reply:
In my testing it was needed if you were not proving the specific group to test against. 
Without it, the credentials that are set in assume_role, are unset before you hit Execute all. So the role assumption works, but the commands do not run as they don't have the required credentials 
This does not happen when you provide a groups of check with -g, as it is executed before the IF. 

Another option would be to leave the If as only referencing [[ $ACCOUNT_TO_ASSUME ]]
but move execute_all and scoring above this so they run with the correct credentials before clean up ?